src/log_batch.rs

```diff
@@ -1174,9 +1250,15 @@ mod tests {
             .unwrap();
         batches.push((batch, Vec::new()));
 
-        for (batch, entry_data) in batches.into_iter() {
-            decode_and_encode(batch.clone(), true, &entry_data);
-            decode_and_encode(batch, false, &entry_data);
+        // for Version::V1
```

Comment:
Use strum::EnumIter and iterate all possible versions instead of handwriting them.